src/main/java/com/hotels/beeju/core/BeejuCore.java

```diff
@@ -91,22 +94,55 @@ public BeejuCore(String databaseName, Map<String, String> preConfiguration, Map<
     configure(preConfiguration);
 
     configureFolders();
-    
+
     configureMetastore();
-    
+
     configureMisc();
 
     configure(postConfiguration);
   }
 
   private void configureMisc() {
+    int webUIPort = getWebUIPort();
+
     // override default port as some of our test environments claim it is in use.
-    conf.setIntVar(HiveConf.ConfVars.HIVE_SERVER2_WEBUI_PORT, 0);
-    
+    conf.setIntVar(HiveConf.ConfVars.HIVE_SERVER2_WEBUI_PORT, webUIPort);
+
     conf.setBoolVar(HiveConf.ConfVars.HIVESTATSAUTOGATHER, false);
-    
+
     // Disable to get rid of clean up exception when stopping the Session.
     conf.setBoolVar(HiveConf.ConfVars.HIVE_SERVER2_LOGGING_OPERATION_ENABLED, false);
+
+    // Used to prevent "Not authorized to make the get_current_notificationEventId call" errors
+    setMetastoreAndSystemProperty(MetastoreConf.ConfVars.EVENT_DB_NOTIFICATION_API_AUTH, "false");
+
+    // Used to prevent "Error polling for notification events" error
+    conf.setTimeVar(HiveConf.ConfVars.HIVE_NOTFICATION_EVENT_POLL_INTERVAL, 0, TimeUnit.MILLISECONDS);
+
+    // Has to be added to exclude failures related to the HiveMaterializedViewsRegistry
+    conf.set(HiveConf.ConfVars.HIVE_SERVER2_MATERIALIZED_VIEWS_REGISTRY_IMPL.varname, "DUMMY");
+    System.setProperty(HiveConf.ConfVars.HIVE_SERVER2_MATERIALIZED_VIEWS_REGISTRY_IMPL.varname, "DUMMY");
+  }
+
+  private void setMetastoreAndSystemProperty(MetastoreConf.ConfVars key, String value) {
+    conf.set(key.getVarname(), value);
+    conf.set(key.getHiveName(), value);
+
+    System.setProperty(key.getVarname(), value);
+    System.setProperty(key.getHiveName(), value);
+  }
+
+  private int getWebUIPort() {
+    // Try to find a free port, if impossible return the default port 0 which disables the WebUI altogether
+    int defaultPort = 0;
+
+    try (ServerSocket socket = new ServerSocket(0)) {
+      return socket.getLocalPort();
+    } catch (IOException e) {
+      log.info(
+          "No free port available for the Web UI. Setting the port to " + defaultPort + ", which disables the WebUI.");
```

Comment:
I think we should also log `e` here as there are a few different reasons the `ServerSocket` constructor can throw an IOE.

Reply:
Makes sense. Added.